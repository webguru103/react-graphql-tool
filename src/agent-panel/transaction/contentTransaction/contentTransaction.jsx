import * as React from 'react';
import IconButton from '@material-ui/core/IconButton';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import Paper from '@material-ui/core/Paper';
import { withStyles } from '@material-ui/core/styles';
import MoreHorizIcon from '@material-ui/icons/MoreHoriz';
import moment from 'moment';
import { compose } from '../../../utility';
import { withCopyTransaction } from '../api/transaction.service';
import c from '../constants';
import DeleteDialog from '../deleteTransactionDialog/deleteTransactionDialog';
import RenameDialog from '../renameTransactionDialog/renameTransactionDialog';
import styles from './contentTransaction.styles';
import { withDialog } from '../../../shared/dialog/withDialog';
import HomeIcon from './ic_home.svg';

type PropType = {
  transactions?: Array<Object>,
  classes?: Object,
  onTransactionClick: (id: string, name: string) => any,
  copyTransaction: (id: string) => any,
  createDialog: ({ dialogContent: React.Element, }) => any,
};

export class ContentTransactionC extends React.PureComponent<PropType> {

  state = {
    anchorEl: null,
    anchorId: null,
    activeTransaction: {},
  };

  handleMenuClick = (e, id) => {
    e.stopPropagation();
    this.setState({
      anchorEl: e.currentTarget,
      anchorId: id,
      activeTransaction: (this.props.transactions.filter(transaction => transaction.id === id))[0],
    });
  };

  handleClose = (e) => {
    e.stopPropagation();
    this.setState({ anchorEl: null, anchorId: null });
  };

  handleRename = (e) => {
    this.handleClose(e);
    this.props.createDialog({
      dialogContent: <RenameDialog transaction={this.state.activeTransaction} />,
    });
  };

  handleDelete = (e) => {
    this.handleClose(e);
    this.props.createDialog({
      dialogContent: <DeleteDialog transaction={this.state.activeTransaction} />,
    });
  };

  handleCopy = (e) => {
    this.handleClose(e);
    this.props.copyTransaction({ id: this.state.anchorId });
  };

  render() {
    const {
      transactions,
      classes,
      onTransactionClick,
    } = this.props;
    const { anchorEl } = this.state;
    return (
      <React.Fragment>
        <Paper classes={{ root: classes.contentHeader }} elevation={0}>
          <span>{c.TABLE_HEAD_NAME}</span>
          <span>{c.TABLE_HEAD_OWNER}</span>
          <span>{c.TABLE_HEAD_LAST_MODIFIED}</span>
        </Paper>
        {transactions && transactions.map(t => (
          <Paper
            key={t.id}
            onClick={() => onTransactionClick(t.id, t.name)}
            classes={{ root: classes.transaction }}
            elevation={1}
          >
            <div className={classes.contentCell}>
              <HomeIcon className={classes.contentTransactionIcon} />
              <span>{t.name}</span>
            </div>
            <div className={classes.contentCell}>
              <span>{t.owner || '-'}</span>
            </div>
            <div className={`${classes.contentCell} ${classes.contentCellWithButton}`}>
              {t.updatedAt ? moment(t.updatedAt).format('MMMM Do YYYY, h:mm A') : '-'}
              <IconButton
                onClick={e => this.handleMenuClick(e, t.id)}
              >
                <MoreHorizIcon />
              </IconButton>
            </div>
          </Paper>
        ))}
        <Menu
          anchorEl={anchorEl}
          open={Boolean(anchorEl)}
          onClose={this.handleClose}
        >
          <MenuItem onClick={this.handleRename}>{c.MENU_RENAME}</MenuItem>
          <MenuItem onClick={this.handleCopy}>{c.MENU_COPY}</MenuItem>
          <MenuItem onClick={this.handleDelete}>{c.MENU_DELETE}</MenuItem>
        </Menu>
      </React.Fragment>
    );
  }

}

ContentTransactionC.defaultProps = {
  transactions: [],
  classes: {},
};

export default compose(
  withStyles(styles),
  withCopyTransaction,
  withDialog,
)(ContentTransactionC);
