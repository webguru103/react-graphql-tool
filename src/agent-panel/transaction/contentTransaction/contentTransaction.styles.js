const styles = (theme: Object) => ({
  contentHeader: {
    backgroundColor: 'transparent',
    display: 'grid',
    marginTop: '20px',
    width: '100%',
    grid: '1fr / 2fr 1fr 1fr',
  },
  contentTransactionIcon: {
    margin: '20px',
  },
  transaction: {
    display: 'grid',
    color: theme.palette.neutrals.secondaryText,
    grid: '1fr / 2fr 1fr 1fr',
    marginBottom: '20px',
    width: '100%',
    cursor: 'pointer',
  },
  contentCell: {
    display: 'flex',
    alignItems: 'center',
  },
  contentCellWithButton: {
    justifyContent: 'space-between',
  },
  tableRowHovered: {
    cursor: 'pointer',
  },
});

export default styles;
