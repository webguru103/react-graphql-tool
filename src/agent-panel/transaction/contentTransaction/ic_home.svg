<?xml version="1.0" encoding="UTF-8"?>
<svg width="24px" height="24px" viewBox="0 0 24 24" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <defs></defs>
    <g id="ic/home" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="ic_home">
            <polygon id="Shape" points="0 0 24 0 24 24 0 24"></polygon>
            <path d="M14,19.5 L14,13.98425 C14,13.71675 13.7522857,13.5 13.4465714,13.5 L10.5534286,13.5 C10.2477143,13.5 10,13.71675 10,13.98425 L10,19.5 L5.42857143,19.5 L5.42857143,9.602 L12,4.5 L18.5714286,9.602 L18.5714286,19.5 L14,19.5 Z" id="Combined-Shape" fill-opacity="0.1" fill="#31303B"></path>
            <polygon id="Stroke-19" stroke="#31303B" stroke-linecap="round" stroke-linejoin="round" points="12 4.5 5.5 9.602 5.5 19.5 18.5 19.5 18.5 9.602"></polygon>
            <polyline id="Stroke-17" stroke="#31303B" stroke-linecap="round" stroke-linejoin="round" points="4 10.75 12 4.5 20 10.75"></polyline>
            <path d="M14,19.5 L10,19.5 L10,13.98425 C10,13.71675 10.2477143,13.5 10.5534286,13.5 L13.4465714,13.5 C13.7522857,13.5 14,13.71675 14,13.98425 L14,19.5 L14,19.5 Z" id="Stroke-21" stroke="#31303B" stroke-linecap="round" stroke-linejoin="round"></path>
            <polygon id="Stroke-23" stroke="#31303B" stroke-linecap="round" stroke-linejoin="round" points="17.5 5 15 5 15 6.76481057 17.5 8.75"></polygon>
        </g>
    </g>
</svg>
