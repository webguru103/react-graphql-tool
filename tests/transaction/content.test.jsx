import React from 'react';
import { ContentTransactionC } from '../../src/agent-panel/transaction/contentTransaction/contentTransaction';
import { shallowWrap, mountWrap } from '../test-helpers/contextWrapper';
import transactionMockData from '../test-helpers/transactionMockData';

describe('ContentC', () => {
  let props;
  let component;
  const wrappedShallow = (lang = 'en') => {
    component = shallowWrap(<ContentTransactionC {...props} />, lang);
    return component;
  };

  const wrappedMount = (lang = 'en') => {
    component = mountWrap(<ContentTransactionC {...props} />, lang);
    component.setProps(props);
    return component;
  };

  beforeEach(() => {
    props = {
      transactions: transactionMockData,
      classes: {},
      onTransactionClick: jest.fn(),
      deleteTransaction: jest.fn(),
      copyTransaction: jest.fn(),
      renameTransaction: jest.fn(),
    };
    if (component) component.unmount();
  });

  test('should render', () => {
    const wrapper = wrappedShallow();
    expect(wrapper).toMatchSnapshot();
  });

  test('should pass in correct props', () => {
    const wrapper = wrappedMount();
    expect(wrapper.find('ContentTransactionC').props().transactions).toBe(transactionMockData);
  });

  test('should call copyTransaction with correct argument', () => {
    const wrapper = wrappedMount();

    expect(wrapper.props().copyTransaction.mock.calls.length).toBe(0);

    // Open menu, click on a button labeled 'Copy'
    wrapper.find('IconButton').first().simulate('click');
    wrapper.find('li').findWhere(n => n.contains('Copy')).first().simulate('click');

    // First argument of first call is the id of first row in table
    expect(wrapper.props().copyTransaction.mock.calls[0][0].id).toBe(transactionMockData[0].id);
    expect(wrapper.props().copyTransaction.mock.calls.length).toBe(1);
  });

});
