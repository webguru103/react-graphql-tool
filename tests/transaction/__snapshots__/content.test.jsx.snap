// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentC should render 1`] = `
<Fragment>
  <WithStyles(Paper)
    classes={
      Object {
        "root": undefined,
      }
    }
    elevation={0}
  >
    <span>
      <FormattedMessage
        defaultMessage="Name"
        id="table-head-name"
        values={Object {}}
      />
    </span>
    <span>
      <FormattedMessage
        defaultMessage="Owner"
        id="table-head-owner"
        values={Object {}}
      />
    </span>
    <span>
      <FormattedMessage
        defaultMessage="Last Modified"
        id="table-head-last-modified"
        values={Object {}}
      />
    </span>
  </WithStyles(Paper)>
  <WithStyles(Paper)
    classes={
      Object {
        "root": undefined,
      }
    }
    elevation={1}
    key="id1"
    onClick={[Function]}
  >
    <div>
      <HomeIcon
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      />
      <span>
        Example 1
      </span>
    </div>
    <div>
      <span>
        -
      </span>
    </div>
    <div
      className="undefined undefined"
    >
      janvier 1er 2018, 12:00 AM
      <WithStyles(IconButton)
        onClick={[Function]}
      >
        <pure(MoreHorizIcon) />
      </WithStyles(IconButton)>
    </div>
  </WithStyles(Paper)>
  <WithStyles(Paper)
    classes={
      Object {
        "root": undefined,
      }
    }
    elevation={1}
    key="id2"
    onClick={[Function]}
  >
    <div>
      <HomeIcon
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      />
      <span>
        Example Deal 2
      </span>
    </div>
    <div>
      <span>
        -
      </span>
    </div>
    <div
      className="undefined undefined"
    >
      janvier 1er 2018, 12:00 AM
      <WithStyles(IconButton)
        onClick={[Function]}
      >
        <pure(MoreHorizIcon) />
      </WithStyles(IconButton)>
    </div>
  </WithStyles(Paper)>
  <WithStyles(Paper)
    classes={
      Object {
        "root": undefined,
      }
    }
    elevation={1}
    key="id3"
    onClick={[Function]}
  >
    <div>
      <HomeIcon
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      />
      <span>
        Example 3
      </span>
    </div>
    <div>
      <span>
        -
      </span>
    </div>
    <div
      className="undefined undefined"
    >
      janvier 1er 2018, 12:00 AM
      <WithStyles(IconButton)
        onClick={[Function]}
      >
        <pure(MoreHorizIcon) />
      </WithStyles(IconButton)>
    </div>
  </WithStyles(Paper)>
  <WithStyles(Paper)
    classes={
      Object {
        "root": undefined,
      }
    }
    elevation={1}
    key="id4"
    onClick={[Function]}
  >
    <div>
      <HomeIcon
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      />
      <span>
        Example 4
      </span>
    </div>
    <div>
      <span>
        -
      </span>
    </div>
    <div
      className="undefined undefined"
    >
      janvier 1er 2018, 12:00 AM
      <WithStyles(IconButton)
        onClick={[Function]}
      >
        <pure(MoreHorizIcon) />
      </WithStyles(IconButton)>
    </div>
  </WithStyles(Paper)>
  <WithStyles(Menu)
    anchorEl={null}
    onClose={[Function]}
    open={false}
  >
    <WithStyles(MenuItem)
      onClick={[Function]}
    >
      <FormattedMessage
        defaultMessage="Rename"
        id="menu-rename"
        values={Object {}}
      />
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      onClick={[Function]}
    >
      <FormattedMessage
        defaultMessage="Copy"
        id="menu-copy"
        values={Object {}}
      />
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      onClick={[Function]}
    >
      <FormattedMessage
        defaultMessage="Delete"
        id="menu-delete"
        values={Object {}}
      />
    </WithStyles(MenuItem)>
  </WithStyles(Menu)>
</Fragment>
`;
